docs/contribute.rst

```diff
@@ -538,6 +538,54 @@ Finally open the svg with any compatible viewer and explore the flamegraph. It w
    :alt: A flamegraph profiling example
    :align: center
 
+rotki Database
+**************
+
+rotki uses two different sqlite databases, one with information about assets (global.db) and one with user information (rotkehlchen.db).
+The latter is encrypted using an extension called `SQLCipher <https://github.com/sqlcipher/sqlcipher>`__
+that provides transparent 256-bit AES full database encryption.
+
+Database Location
+=================
+
+Databases are stored in directories under the `rotki data directory <https://rotki.readthedocs.io/en/latest/usage_guide.html#rotki-data-directory>`__.
+
+The global database is stored at ``global_data/global.db``.
+
+The accounts you create in rotki have their own database stored at ``<account_name>/rotkehlchen.db``.
+
+Exploring the database
+======================
+
+To open the user database you can use `sqlitebrowser <https://sqlitebrowser.org/>`__.
+It supports sqlcipher and will ask for the password used to decrypt the database.
+
+If you prefer the command line instead, you can use the ``sqlcipher`` cli tool. Note: using just ``sqlite3`` cli will not work since the database is encrypted.
+
+Note to Debian and Ubuntu users: we are using SQLCipher encryption v4, therefore a recent version of sqlcipher is required. Unfortunately
+the version available in the distribution repositories are too old and won't let you open the database using ``sqlitebrowser`` nor ``sqlcipher``.
+On top of that, the ``sqlitebrowser`` version provided is not compiled with sqlcipher support.
+As a workaround, you can find a `PPA <https://launchpad.net/ubuntu/+ppas>`__ **(use at your own risk)** to
+install more recent versions of both packages and you can also recompile sqlitebrowser with sqlcipher support following this
+`stackoverflow thread <https://stackoverflow.com/questions/48105035/sqlite-browser-without-sqlcipher-support-in-ubuntu>`__.
+
+When using sqlcipher, you need to specify the password to decrypt the database entering ``PRAGMA key='your-secret-key';`` right after opening the database.
+
+Migrations
```

Comment:
So here I think you would need to go even deeper.

We have two different things for the user DB:
1. Is the DB migrations. These are here: https://github.com/rotki/rotki/tree/develop/rotkehlchen/data_migrations. They are only for migrating data and that means the database schema does not change. Can also be something as simple as deleting backup files or something. The important thing here is the schema **does not** change and as such you can easily open the DB with the previous version of rotki.
2. The DB Upgrades, which is what you call here migrations. This is indeed changing the schema and it's important to note here in the docs that it's not reversible. In order to open the upgraded database with an older version you would need to have had a backu.p.


Also it would be worth mentioning that the global DB also has the same thing with upgrades (we have had no simple data migrations there yet)

Reply:
Thanks! I just added it. I'm learning a lot :smiley: